doc/source/whatsnew/v0.25.0.rst

```diff
@@ -204,7 +204,7 @@ Interval
 Indexing
 ^^^^^^^^
 
--
+- Added an exception message when calling :meth:`DataFrame.iloc` with a list of non-numeric objects.
```

Comment:
Can you add a reference to the issue number here?

Reply:
FIxed